main.go

```diff
@@ -723,7 +723,9 @@ func (tf *transformer) handleDirectives(comments []*ast.CommentGroup) {
 			// This directive has two arguments: "go:linkname localName newName"
 
 			// obfuscate the local name.
-			fields[1] = hashWith(curPkg.GarbleActionID, fields[1])
+			if curPkg.Private {
```

Comment:
is this the only "is private" check we've got left in the obfuscation logic? if so, I'd document why it needs to be here.

Reply:
I think so yes